### PR TITLE
Fix bug in config.ForPlugin and add tests

### PR DESCRIPTION
**Pull request by edwarnicke**
config.ForPlugin was failing to work properly
in the case of no opts...

This patch fixes it, and introduces tests to insure proper
operation

Signed-off-by: Ed Warnicke <hagbard@gmail.com>

### CODE REVIEW

File: config/config_test.go [hunk 1]
```diff
@@ -0,0 +1,53 @@
+// Copyright (c) 2017 Cisco and/or its affiliates.
```
Comment (ondrej-fabry) [inline config/config_test.go]: It's 2018.

File: config/config_test.go [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package config_test
+
+import (
+	"testing"
+
+	"github.com/ligato/cn-infra/config"
+	"github.com/onsi/gomega"
```
Comment (ondrej-fabry) [inline config/config_test.go]: Use dot import for more readable tests. 
`Expect(pluginConfig).ShouldNot(BeNil())` vs. `gomega.Expect(pluginConfig).ShouldNot(gomega.BeNil())`

File: config/config_test.go (hunk 1 above)
Comment (edwarnicke) [reply config/config_test.go]: Done

File: config/config_test.go (hunk 2 above)
Comment (edwarnicke) [reply config/config_test.go]: Done

**Pull request merged**